### ISSUE TITLE
Amplify fails reading profile since 4.48.0

### ISSUE THREAD

**Issue by bashtoni** [aws-amplify/amplify-cli#7311]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

yarn

### If applicable, what version of Node.js are you using?

14.16.1

### Amplify CLI Version

4.50.2

### What operating system are you using?

Fedora 34

### Amplify Categories

Not applicable

### Amplify Commands

push

### Describe the bug

After upgrading to 4.50.2, when running `amplify push` I see the following error:

```
Fetching updates to backend environment: dev from the cloud.Failed to get profile: Profile configuration for 'foo_master' is invalid: missing aws_access_key_id, aws_secret_access_key
```

The relevant (redacted) sections from my `~/.aws/config` file are below.  `foo-dev` is the profile that should be used:

```
[profile foo_master]
aws_access_key_id = AKIAredacted
aws_secret_access_key = redacted
region = us-west-2
mfa_serial = arn:aws:iam::123456789012:mfa/sam

[profile foo-dev]
role_arn = arn:aws:iam::012345678901:role/OrgDevRole
source_profile = foo_master
region = us-west-2
```

I have confirmed my profile is working correctly for AWS CLI:
```
$ aws sts get-caller-identity --profile foo-dev
{
    "UserId": "AROAJC4UWCDKB66RTDPJY:botocore-session-1620771173",
    "Account": "012345678901",
    "Arn": "arn:aws:sts::012345678901:assumed-role/OrgDevRole/botocore-session-1620771173"
}
```

After trying each prior version of Amplify CLI, I discovered that everything works as expected on 4.47.1.

### Expected behavior

Correctly read profile.

### Reproduction steps

Follow the steps in the bug report when using 4.48.0 or later.

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by gitandrewk** [comment]
This happened to me too. My problem was fixed because there was a trailing space after the profile name

Ex: [profile foo_master]*space*

It was a couple weeks back when I was troubleshooting, but when I was digging through the cli code, I saw a change where I believe a try/catch was added which "broke" the deployment for me.

Also, I think the source of the problem is the ini package, so potentially not an Amplify problem.

**Comment by bashtoni** [comment]
Thanks.  I checked and there are no trailing spaces anywhere in my `~/.aws/config` file, so it seems like this is potentially a different problem.

**Comment by eioo** [comment]
I got this error message today too when trying to run `amplify pull`. No trailing spaces or such in the credentials file. Downgrading to version 4.47.1 solved the issue. I'm on OSX.

**Comment by marcvberg** [comment]
Hi All,

I have a fix for the issue and PR \#7401 is approved, we're just waiting on it to get pulled into a release. The issue is showing up when source_profile is used, so you may be able to work around that temporarily by using a profile which has the credentials included in it, such as the profile example foo_master above.

Thank you for your patience!
